endpoint: move most bpf, bpf map dependencies into bpf.go

ianvernon opened the pull request:
The code dealing with bpf-related functionality should be
moved into a file that utilizes bpf-related code, bpf.go.

No functional changes are intended by this. By logically collocating code which is related to the datapath, it makes it easier for us to potentially abstract out the datapath in `pkg/endpoint` in the future.

Signed-off by: Ian Vernon <ian@cilium.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6205)
<!-- Reviewable:end -->

Commit:
endpoint: move most bpf, bpf map dependencies into bpf.go

The code dealing with bpf-related functionality should be
moved into a file that utilizes bpf-related code, bpf.go.

Signed-off by: Ian Vernon <ian@cilium.io>

joestringer commented on pkg/endpoint/endpoint.go:
```diff
@@ -41,9 +40,6 @@ import (
-	"github.com/cilium/cilium/pkg/maps/cidrmap"
-	"github.com/cilium/cilium/pkg/maps/ctmap"
-	"github.com/cilium/cilium/pkg/maps/lxcmap"
 	"github.com/cilium/cilium/pkg/maps/policymap"
```
Is there something different about policymap that prevents removing this dependency? I guess we're building up maps of the actual datapath value representation from that package.

ianvernon replied:
The endpoint structure contains a reference to `policymap.PolicyMap`.

aanm commented on pkg/endpoint/bpf.go:
```diff
@@ -59,6 +62,50 @@ const (
 )
 
+// mapPath returns the path to a map for endpoint ID.
+func mapPath(mapname string, id int) string {
```
Can you make a commit that makes this function a method of `Endpoint`?

```golang
func (e *Endpoint) mapPath(mapName string) string {
	return bpf.MapPath(mapname + strconv.Itoa(e.ID))
}
```

ianvernon replied:
Done.

aanm reviewed (changes requested):
Only a small nit

Commit:
endpoint: change mapPath to have an endpoint receiver

Signed-off by: Ian Vernon <ian@cilium.io>

aanm commented on pkg/endpoint/bpf.go:
```diff
@@ -59,6 +62,50 @@ const (
+
+// CallsMapPathLocked returns the path to cilium tail calls map of an endpoint.
+func (e *Endpoint) CallsMapPathLocked() string {
+	return CallsMapPath(int(e.ID))
```
`return bpf.MapPath(CallsMapName + strconv.Itoa(id))`

aanm commented on pkg/endpoint/bpf.go:
```diff
@@ -59,6 +62,50 @@ const (
+	return bpf.MapPath(PolicyGlobalMapName)
+}
+
+func CallsMapPath(id int) string {
```
This function is not used anywhere besides line 106

aanm reviewed (dismissed):
Last small nits

Commit:
endpoint: remove `CallsMapPath` function

This was no longer used outside of `CallsMapPathLocked`; factor the code from
`CallsMapPath` into `CallsMapPathLocked`.

Signed-off by: Ian Vernon <ian@cilium.io>
